DataHandler.py

```diff
@@ -0,0 +1,60 @@
+import os
+import yaml
+import time
+from pathlib import Path
+
+class DataHandler:
+    def __init__(self, data_directory="data"):
+        self.data_directory = data_directory
+
+        if not os.path.exists(self.data_directory):
+            os.mkdir(self.data_directory)
+
+    def readData(self, data: list):
+        # TODO I'm assuming the data is in a certain order, is there a better way or do I just need to make this assumption?
```

Comment:
The data coming in here is from the QR being read, right? So if that's an array/list of strings or something, then there isn't much you can do other than simply assume the order. Otherwise you'd need to implement some different way of encoding the data coming out of the QR codes. For example, if the QR code is encoding the values of a form, then rather than simply encoding this as the array `["value1", "value2"]` it could be encoded in JSON as `{"key1":"value1", "key2":"value2"}`. If whatever mechanism you have on the scouting app doesn't have a way to encode data as a JSON object or map/dict structure like that then some kind of parseable encoding of the values could work as well, ex. `["key1;value1", "key2;value2"]`, where the semicolons are used as `key;value` separators (and the assumption is that the semicolon is a character which could not appear in the key or the value - perhaps in this case the keys and/or values should be URI encoded).

Reply:
Or, an even more primitive but simple and reliable way: `["key1", "value1", "key2", "value2"]` - simply an array where each member is a form key iff its index is even, and odd indices encode form values for the key of the preceding index.

Reply:
Btw, this is what I was talking about on our call previously when I talked about making this system data-agnostic - TBA API and the scouting app can be game-specific, but this Odin backend should be as agnostic as possible and not know anything about the contents of the documents. Certain fields are fine to assume exist, if need be, such as the Scouter Name, Match Number, Team Number, Alliance Colour, etc., but fields like "High Port Auto" or "Color Wheel Done" are very game-specific and should ideally not appear anywhere in Odin.

Reply:
I think I said in Slack at some point that I'll eventually encode the data in json/yaml within the actual scouting app itself, whether that is the existing one, or a rewritten one. Alternatively, alternating between keys/values or encoding them a different way is another good idea.

Reply:
But yea, as of now there really isn't any way to make this agnostic. However, a future focus will be to either transmit game-agnostic data or just do the encoding in json/yaml within the app itself.

Reply:
One advantage of encoding it with yaml within the app is instead of having a function/method to convert whatever format we choose into yaml, instead we can instead basically just have something like (in pseudocode)
```
func write_data(in_data): 
 open file f
 write data to file f
```
instead of having to add unnecessary steps for ourselves.